core/src/banking_stage/decision_maker.rs

```diff
@@ -0,0 +1,190 @@
+use {
+    crate::leader_slot_banking_stage_metrics::{LeaderSlotMetricsTracker, MetricsTrackerAction},
+    solana_poh::poh_recorder::{BankStart, PohRecorder},
+    solana_sdk::{
+        clock::{
+            DEFAULT_TICKS_PER_SLOT, FORWARD_TRANSACTIONS_TO_LEADER_AT_SLOT_OFFSET,
+            HOLD_TRANSACTIONS_SLOT_OFFSET,
+        },
+        pubkey::Pubkey,
+    },
+    std::sync::RwLock,
+};
+
+#[derive(Debug, Clone)]
+pub enum BufferedPacketsDecision {
+    Consume(BankStart),
+    Forward,
+    ForwardAndHold,
+    Hold,
+}
+
+pub struct DecisionMaker;
```

Comment:
nit: probably can make the name more explicit

Reply:
I think right now given the context it is fine. I have considered renaming the `BufferedPacketsDecision` to `SchedulingDecision` or something similar. Would at the same time rename this to `SchedulingDecisionMaker`, but will leave it to a separate PR if we end up doing that.